mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/ISyncTargetNode.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.mpsadapters
+
+import org.modelix.model.api.IChildLink
+import org.modelix.model.api.INode
+
+interface ISyncTargetNode : INode {
+    /**
+     * Some implementations of INode may not be able to store arbitrarily structured models, but are adapters to other
+     * data structures that may need more information than just the concept to instantiate the child. Often this is just
+     * a property such as the name.
+     *
+     * This function allows providing that required information. In case of a synchronization the sync algorithm just
+     * passes the node of the source model that it tries to copy. The INode is allowed to copy as much of the
+     * [templateNode] as it wants. The caller should not assume any more being copied as if it just provided the
+     * concept, but also expect the whole subtree being copied.
+     *
+     * Another valid use case for this method might be bidirectional model transformations. It's similar to a simple
+     * model synchronization in these sense that there two models where changes in one are applied to the other side,
+     * but part of the synchronization needs to be delegated to the INode implementation.
+     */
+    fun addNewChild(role: IChildLink, index: Int, templateNode: INode): INode
+}
+
+fun INode.addNewChild(role: IChildLink, index: Int, templateNode: INode): INode {
```

Comment:
I think, we should not add this overload.
By accident users can make the wrong import and call the wrong functionality.

Reply:
The implementation was incomplete and is fixed now.